docs/migration-guide-v2.md

```diff
@@ -0,0 +1,214 @@
+# Parser v1 to v2 Migration Guide
+
+## General
+
```

Comment:
I am missing a bit of context why I am reading a list here.

Reply:
Maybe we will change that header to the `tl;dr`? 😆 Because it's for that.

Reply:
Would make sense yes 😄